dali/kernels/imgproc/jpeg/jpeg_distortion_gpu_test.cu

```diff
@@ -14,67 +14,86 @@
 
 #include <cuda_runtime_api.h>
 #include <gtest/gtest.h>
+#include <opencv2/opencv.hpp>
 #include <vector>
+#include <cstdlib>
 #include "dali/core/cuda_event.h"
 #include "dali/core/cuda_stream.h"
+#include "dali/pipeline/data/tensor_list.h"
 #include "dali/test/tensor_test_utils.h"
+#include "dali/test/test_tensors.h"
+#include "dali/test/dali_test_config.h"
 #include "dali/kernels/test/kernel_test_utils.h"
 #include "dali/kernels/imgproc/jpeg/jpeg_distortion_gpu.cuh"
 
 #define DEBUG_LOGS 0
-#define PERF_RUN 0
 
 namespace dali {
 namespace kernels {
 namespace test {
 
 using KernelPtr = void(*)(const SampleDesc *, const kernels::BlockDesc<2> *);
+using testing::dali_extra_path;
 
 template <typename GTestParams>
 class JpegDistortionTestGPU : public ::testing::Test {
   using T = typename GTestParams::T;
   static constexpr bool vert_subsample = GTestParams::vert_subsample;
   static constexpr bool horz_subsample = GTestParams::horz_subsample;
+  static constexpr bool use_real_images = true;
+  static constexpr bool perf_run = false;
+  static constexpr bool dump_images = false;
 
  public:
-  JpegDistortionTestGPU() {
-    in_shapes_ = {{200, 400, 3}, {2000, 20, 3}, {2, 2, 3}};
-#if DEBUG_LOGS
-    in_shapes_ = {{2, 4, 3}, {2, 2, 3}};
-#elif PERF_RUN
-    in_shapes_.resize(64, TensorShape<3>{600, 800, 3});
-#endif
-    input_host_.resize(batch_volume(in_shapes_));
-  }
-
   void SetUp() final {
-    std::mt19937_64 rng;
-    UniformRandomFill(input_host_, rng, 0., 255.);
-
-    int64_t nbytes = input_host_.size() * sizeof(uint8_t);
-    CUDA_CALL(cudaMalloc(&input_device_, nbytes));
-    CUDA_CALL(cudaMemcpy(input_device_, input_host_.data(), nbytes, cudaMemcpyDefault));
-
-    // for simplicity we are allocating 3 planes of the same size, even though
-    // we only need half or a fourth of the memory, due to subsampling.
-    CUDA_CALL(cudaMalloc(&output_, nbytes));
-    CUDA_CALL(cudaMemset(output_, 0, nbytes));
-    cudaDeviceSynchronize();
+    if (use_real_images) {
+      std::vector<std::string> paths{
+        dali_extra_path() + "/db/single/bmp/0/cat-111793_640_palette_8bit.bmp",
+        dali_extra_path() + "/db/single/bmp/0/cat-1046544_640.bmp",
+        dali_extra_path() + "/db/single/bmp/0/cat-3591348_640.bmp",
+      };
```

Comment:
```suggestion
      std::vector<std::string> path = ImageList(testing::dali_extra_path() + "/db/single/bmp", {".bmp"}, 3);
```
I think it is better to abstract away particular file names. Unless they have a properties that are a must for this test.